test/sqllogictest/types.slt

```diff
@@ -963,3 +953,95 @@ NULL
 # ensure that all existing types can be cast to their respective names
 statement OK
 select oid, oid::regtype::text from (select oid from mz_catalog.mz_types)
+
+# 🔬🔬 regclass
+
+query T
+SELECT 1::regclass
+----
+1
+
+query T
+SELECT 1::int4::regclass
+----
+1
+
+query T
+SELECT 1::oid::regclass
+----
+1
+
+query T
+SELECT 1::oid::regclass::oid
+----
+1
+
+query T
+SELECT '1'::regclass
+----
+1
+
+query T
+SELECT '1'::pg_catalog.regclass
+----
+1
+
+query T
+SELECT '1'::regclass::text
+----
+1
+
+query T
+SELECT 'mz_tables'::regclass::text
+----
+mz_tables
+
+query T
+SELECT 'mz_tables'::regclass
+----
+20162
```

Comment:
I don't know offhand how table IDs are generated. Is this safe?

Reply:
I know that my instance of Materialize always gets a different ID, but the test always gets the same one.

Reply:
Ok, this appears to be a function of the iteration order and allocate_oid: https://github.com/MaterializeInc/materialize/blob/10aac0fd020f313ed71505219189bdfdb2d1189e/src/coord/src/catalog.rs#L938

and my spidey sense is that this will cause test flakes in the future. How about:

```
SELECT 'mz_tables'::regclass = (SELECT oid FROM mz_objects WHERE name = 'mz_tables')
```